### ISSUE TITLE
[Bug Report] v-menu doesnt close when menu item has a dialog

### ISSUE THREAD

**Issue by mrodal** [vuetifyjs/vuetify#11521]
### Environment
**Vuetify Version:** 2.2.30
**Vue Version:** 2.6.11
**Browsers:** Chrome 83.0.4103.61
**OS:** Windows 10

### Steps to reproduce
On the reproduction link click on the menu item

### Expected Behavior
dialog opens and menu closes

### Actual Behavior
dialog opens and menu **stays open**

### Reproduction Link
<a href="https://codepen.io/mrodal/pen/JjYQvOE" target="_blank">https://codepen.io/mrodal/pen/JjYQvOE</a>

**Comment by standup75** [comment]
Workaround:

```
<script type="text/x-template" id="app-template">
  <v-app>
    <v-container>
      <v-menu bottom offset-y v-model="menu">
            <template v-slot:activator="{ on }">
              <v-btn color='primary' v-on="on">menu</v-btn>
            </template>

            <v-list dense>
                  <v-list-item @click.stop="open()">
                    <v-list-item-title >Open dialog</v-list-item-title>
                  </v-list-item>
              <v-dialog v-model="dialog" width="500">

                <v-card>
                  <v-card-title>Dialog</v-card-title>
                </v-card>
              </v-dialog>
            </v-list>
          </v-menu>
    </v-container>
  </v-app>
</script>

<div id="app"></div>
```

```
// Looking for the v1.5 template?
// https://codepen.io/johnjleider/pen/GVoaNe

const App = {
  template: "#app-template",
  data: () => ({
    dialog: false,
    menu: false,
  }),
  methods: {
    hi() {
      alert("hi");
    },
    open() {
      console.log('//')
      this.dialog = true;
      this.menu = false;
    }
  }
};

new Vue({
  vuetify: new Vuetify(),
  render: (h) => h(App)
}).$mount("#app");

```

**Comment by Joebayld** [comment]
Would love to see this fixed! I have the same issue and the workaround adds unnecessary extra components when working with a large project.

**Comment by KaelWD** [comment]
In v3 we now unmount menu content when it's closed (#6764). Would you be ok with the menu scheduling itself to close after the dialog is closed instead?

**Comment by mrodal** [comment]
Hmm its better, but still is not exactly the expected behaviour. Would it be possible to do it the other way around? closing the menu and scheduling the unmount to after closing the dialog?

**Comment by KaelWD** [comment]
I might have to play with that a bit, currently unmounting is tied directly to visibility.

**Comment by KaelWD** [comment]
There's also cases where you'd want the parent item to stay open, for example nested menus or a menu in a drawer.

**Comment by mrodal** [comment]
Thats true.. dialog its a special case.. Maybe its a thing with the dialog design, the feeling I get is that the dialog activator should be completely detached from the actual dialog.. and it should be like clicking a link, unmounting the component that had the link doesnt close the new tab. But I guess for that we would have to do something like the workaround

So your solution is good enough for me given the design, thanks!

**Comment by Fahmiin-Abdullah** [comment]
> Would love to see this fixed! I have the same issue and the workaround adds unnecessary extra components when working with a large project.

For now, simplest and cheapest way would be to add `refs` in the `v-menu` and after closing dialog, trigger `this.$refs.menu.isActive = false`

**Comment by Seblor** [comment]
Same problem, I had to use a v-model on the v-menu and manually set it to false when handling the click on the v-dialog.

This needs to be resolved as it means the components of the same library do not play fair with each others.

**Comment by zub0r** [comment]
The menu can be closed [modifying the menu model on click](https://play.vuetifyjs.com/#eNqtVMFu2zAM/RVCl7ZAZa/rdjHSoMmw43bbqe5BsZVMqywJkuI1CPzvoyTbcZNtyaEnW9TjeyRF8mlPFsZk7ZaTgsw8b4xkns9LBTCrRQuVZM49lMTzV08rrjy30DB696EkEZRg84arbbHfQ/hC183yYOzvWxqt+NE1l8gVjqM7AgZVhDipfcEqL1rmtUXsHozVxkE3cYicK6+gZVYw5RGmt14KxeuSIMlKqBpt0RHd4NvX7z9glkefg2o+SXY0tlQK598qBQsViB6LYRg95N/jKt0YeKykqF76FOEB1kw6jsjFYjHLA2LKjAGN1O+uuFwuL1QcrMNz4TFwpRbo3/GoWDNXWWGSh2iMth6+hGDWVjdwleXhEFrqKlWWv0ZIzddsKz3sk1AITivsKFcMJog0t+nQ9d+aeVbA9Q08zOF6BMaGS9kO+Jv404Vox/jILcG+eHHZL6cVdnj0L0nlsDEKeEqeJfnpvXFFnle1QiR2qWhtprjPlWlyTMSL9e7xPrvP7j5jRZwfbFSylcsiW6B6LlWHiqkktGHmSDZdROk+kZIgUzifieERYbndKi8aTmvdxGA+pVgm5oy7hq6s/u24RZKS9LWJMiHgC6TOposaJ9w5CrbcUstVzS23l6Z05DZN6+jqJLX+sUPBh347XWEtrQWTejM0939XTVwY00UTN+DxPpnO5EdYbejGcq7erqfADflkxCbrcDpKo8jAiXTGioZZfKl5Ch1Wut4dGMJ8jjn9fS5BMrUJOztsv2D/9xAeD9eZ8TodsCDqd5IPmi4OwxwRwRomsLsl8VXxnVJLkec/Rw8TBw==), but it looks weird, because it closes the menu only after the dialog is closed. 
Any chance the auto closing of the menu can be fixed for v3?

**Comment by kkyryl** [comment]
This seems to travel from milestone to milestone but I wanted to support what others have said — it's a common use case, a workaround requires a lot of unnecessary code and is close to impossible in nested components. A proper solution from the user experience perspective would be to close the menu while keeping the dialog open, but any solution which keeps the dialog open without having to jump through the nested hoops to juggle all the open menus would be great. Thank you!

**Comment by Kroeg93** [comment]
Still an issue - here is another workaround: 

You could try to wrap your list elements within 
`<template v-slot:default="{isActive}">`

and then use the @ click event on your list elements like:
`@click="isActive.value = false"`

This works for me and closes v-menu, while my dialog is still open.

**Comment by J-Sek** [comment]
With this structure, it might be a bit tricky to "just close" the menu.
- [example](https://play.vuetifyjs.com/#eNp1U9FO4zAQ/JWV74E7qUm5OyFBVRBIvPMBhAc32YAlx7bsTRCq+u+s7RbSJLwlu+Px7Hj2eS+Cr9cPzpVDj2IjtoSd05LwrjIA26GQzqXP9FNbQ1IZ9MdSKnZoethZItuBbduAVHwAytcRinEnXhiKoC1tZE1qkGT9bSX24Lx1YQPWwKESo2Ppgh0ZqK1OUOdVJ/1HJZhnp0zDJWv4SBSxXSfs+Nb1aJwlMb8abGWvKYlQ4SGKwiUNWgWCBk3Ixsx6hWJO1uSVc3qKOUcVpIghTw4NNEpq+xqFT9sLDBk8bQCMrXTSo6FKzFHvqqE3RlxdXi6172VL6As+jJHn5EU5SN0j3EIrdcD5wZnOnBPpm4XOV+844+PX9KPiAuERMPeEG9mU6aOMDT1/ymPnp5B8o2KiTsmPAs6iHyt5MyYR24aaI0DAW9DnxVGds55gDx5bOEDreU0ueNkuciKZNyYrTcE2M+h3svpPpM5kTCMOK/G/vCr/Xov4cVP+E6sEW53tb669fAKCTx5q) uses `@after-enter` to close menu
  - requires `eager` to avoid accidentally closing the dialog
  - includes additional `v-ripple` to have any visual feedback upon click

Alternatively, `@after-leave` might help close he menu right after the dialog disappears (does not require `eager`)

**Comment by juanjcardona13** [comment]
Workaround:

[here](https://play.vuetifyjs.com/playgrounds/KCreoQ)

Is it coming closer?

**Comment by juanjcardona13** [comment]
After thorough investigation, this issue should be **closed as "not a bug"** or **"working as intended"**. The reported behavior is actually the correct and expected behavior based on Vuetify's architectural design 

## Technical Analysis

### Root Cause Analysis

The issue occurs due to a fundamental conflict in event handling priorities when a `Element` serves dual roles:

1. **Content element** of `VMenu` (should trigger menu close on click)
2. **Activator element** of `VDialog` (should trigger dialog open on click)

### Code Analysis

#### Event Handling Priority Conflict

Activator events take priority

// Lines 108-115: Activator events (dialog open logic)

https://github.com/vuetifyjs/vuetify/blob/4eeb514c675b2eeef16f994cac7d37fe1a516b65/packages/vuetify/src/components/VOverlay/useActivator.tsx#L108-L115

vs

// Lines 187-193: Content events (menu close logic)

https://github.com/vuetifyjs/vuetify/blob/4eeb514c675b2eeef16f994cac7d37fe1a516b65/packages/vuetify/src/components/VOverlay/useActivator.tsx#L187-L193

The `e.stopPropagation()` in the activator event handler (line 109) prevents the content click event from reaching the menu's close logic.

#### Content Lifecycle Management

In

https://github.com/vuetifyjs/vuetify/blob/4eeb514c675b2eeef16f994cac7d37fe1a516b65/packages/vuetify/src/components/VOverlay/VOverlay.tsx#L295

This behavior is intentional and was implemented as part of [Feature Request #6764](https://github.com/vuetifyjs/vuetify/issues/6764) to clean up bootable content on close.

_**This means that even if you manage to close the VMenu, the VDialog will also be closed/removed (unless you use "eager," as explained below), since it is part of the VMenu's content.**_

## Recommended Solutions

### Solution 1: Proper Architecture (Recommended)

```vue
<template>
  <div class="pa-4">
    <VMenu>
      <template #activator="{ props }">
        <VBtn v-bind="props" color="primary" text="Open Menu"/>
      </template>
      <VList>
        <VListItem @click="dialogOpen = true" title="Open Dialog" />
        <VListItem title="Another Item" />
      </VList>
    </VMenu>

    <!-- Dialog outside of menu -->
    <VDialog v-model="dialogOpen">
      <VCard>
        <VCardTitle>Test Dialog</VCardTitle>
      </VCard>
    </VDialog>
  </div>
</template>

<script setup>
import { ref } from 'vue'
const dialogOpen = ref(false)
</script>
```

### Solution 2: Using Eager Mode (If Dialog Must Be Inside Menu) by @J-Sek 

> With this structure, it might be a bit tricky to "just close" the menu.
> 
> * [example](https://play.vuetifyjs.com/#eNp1U9FO4zAQ/JWV74E7qUm5OyFBVRBIvPMBhAc32YAlx7bsTRCq+u+s7RbSJLwlu+Px7Hj2eS+Cr9cPzpVDj2IjtoSd05LwrjIA26GQzqXP9FNbQ1IZ9MdSKnZoethZItuBbduAVHwAytcRinEnXhiKoC1tZE1qkGT9bSX24Lx1YQPWwKESo2Ppgh0ZqK1OUOdVJ/1HJZhnp0zDJWv4SBSxXSfs+Nb1aJwlMb8abGWvKYlQ4SGKwiUNWgWCBk3Ixsx6hWJO1uSVc3qKOUcVpIghTw4NNEpq+xqFT9sLDBk8bQCMrXTSo6FKzFHvqqE3RlxdXi6172VL6As+jJHn5EU5SN0j3EIrdcD5wZnOnBPpm4XOV+844+PX9KPiAuERMPeEG9mU6aOMDT1/ymPnp5B8o2KiTsmPAs6iHyt5MyYR24aaI0DAW9DnxVGds55gDx5bOEDreU0ueNkuciKZNyYrTcE2M+h3svpPpM5kTCMOK/G/vCr/Xov4cVP+E6sEW53tb669fAKCTx5q) uses `@after-enter` to close menu
>   
>   * requires `eager` to avoid accidentally closing the dialog
>   * includes additional `v-ripple` to have any visual feedback upon click
> 
> Alternatively, `@after-leave` might help close he menu right after the dialog disappears (does not require `eager`)

```vue
<template>
  <v-menu eager>
    <template v-slot:activator="{ props: on }">
      <v-btn color="primary" v-bind="on" text="menu" />
    </template>

    <template #default="{ isActive }">
      <v-list dense>
        <v-list-item>
          <v-list-item-title>Open dialog</v-list-item-title>
          <v-dialog
            activator="parent"
            width="500"
            @after-enter="isActive.value = false"
          >
            <v-card>
              <v-card-title>Dialog</v-card-title>
            </v-card>
          </v-dialog>
        </v-list-item>
      </v-list>
    </template>
  </v-menu>
</template>
```


## Comparison with Other Libraries

This behavior isn't unique to Vuetify; it also occurs with other libraries, and it behaves similarly if you nest a Dialog component inside a Menu.
This suggests that this behavior is not a "bug" specific to Vuetify.

## Conclusion

The reported behavior is **working as intended**.

### Recommendation

**Close this issue** with the following resolution:

1. **Status**: `wontfix` or `not a bug`
2. **Reason**: Working as intended - architectural design choice
3. **Action**: Update documentation with proper usage patterns
4. **Reference**: Provide the recommended solutions above

### Documentation Update Suggestion

Consider adding a section to the VMenu documentation about:
- Proper component composition patterns
- When to use `eager` mode
- Best practices for dialogs in menus

This will help developers avoid similar confusion in the future.

@KaelWD @mrodal

**Comment by Glandos** [comment]
I have a side use case, with VOverlay, that is **`contained`** within the VMenu content: https://play.vuetifyjs.com/#eNptUsFuwjAM/RUvlw0JWsHEpSpoHHeYdthx3SEkRlSkTZWajgnx73OSlhW0m/P8/Pz82s+zaJ1KN02TdEcUmcgJq8ZIwnVRA+TdTO1RHbb2BN2sshrNqhC2Q2fkTyHAyG1ANlpDj0JZQ4X1sRDrPP0bj3K67EAZ2bY8Q3iimcKa0DHXt8O+LdWgrLGOKeic5Sa8KFOqAwNeGFbw0C+IQwAfe/sNbx7rAdYJVKmo7CQFsUY6XjaaiudJp4FKMsiUIDEmBMpw2H8BKFuTLGvUkDXo2rIl3sGMnTQt3kkBvD5WPp5+3MczKI1oY38+QDZ4hTzgDxvi4ifnFbNNOVyu8nT0BfnZKlc2FCh4aqwj0LiTR0NwjiJakszgaQKrNTz1GIDU+j16yyAcMx06veV72Nu6xS6TUFy8p6sLcZmK52SZLJbCF/N5shDTOHXzJ0bs6xcl59HN

If the overlay is activated, then the menu is not closed on click outside, even if the overlay is contained within the menu content.

Is it the same issue? Or should I open a new one?

**Comment by J-Sek** [comment]
It is something else. (workaround: `v-click-outside="{ handler: () => showMenu = false }"` on the VCard)
